### ISSUE TITLE
segfault when colleague shares their screen after chromium 99 update

### ISSUE THREAD

**Issue by christianfosli** [flathub/org.chromium.Chromium#207]
After update to chromium 99.0.4844.51 the browser has started segfaulting most of the time when my colleague shares their screen in a Microsoft Teams call. It causes the tab to crash showing the "Aw, snap!" page and SIGSEGV error.

I'm not sure if it can be related to the flatpak runtime / permissions or chromium itself, and not really sure how to debug it further.
It happens both if I run chromium with ozone-x11 backend (XWayland) and native wayland (`--ozone-platform-hint=auto`).

<details>
<summary>Related logs from journalctl around the time of the crash</summary>

```
Mar 04 11:35:34 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:8531 error:1 (Permission denied)
Mar 04 11:35:35 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:8535 error:1 (Permission denied)
Mar 04 11:35:36 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:8556 error:1 (Permission denied)
Mar 04 11:35:37 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:8560 error:1 (Permission denied)
Mar 04 11:35:44 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:8615 error:1 (Permission denied)
Mar 04 11:35:45 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:8619 error:1 (Permission denied)
Mar 04 11:35:49 fedora audit[52513]: ANOM_ABEND auid=1000 uid=1000 gid=1000 ses=3 subj=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 pid=52513 comm="WebRTC_Network" exe="/app/chromium/chrome" sig=11 res=1
Mar 04 11:35:49 fedora audit: BPF prog-id=301 op=LOAD
Mar 04 11:35:49 fedora audit: BPF prog-id=302 op=LOAD
Mar 04 11:35:49 fedora audit: BPF prog-id=303 op=LOAD
Mar 04 11:35:49 fedora systemd[1]: Started Process Core Dump (PID 92650/UID 0).
Mar 04 11:35:49 fedora audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-coredump@2-92650-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Mar 04 11:35:52 fedora systemd-coredump[92651]: Process 52513 (chrome) of user 1000 dumped core.
                                                
                                                Found module /app/chromium/chrome with build-id: 5103830f64867e19a62bede783d4c71663d46a10
                                                Found module /usr/lib/x86_64-linux-gnu/libbrotlicommon.so.1.0.9 with build-id: 5fb51aa2b479def09f104c44c0a126e22a221f7a
                                                Found module /usr/lib/x86_64-linux-gnu/libgpg-error.so.0.32.2 with build-id: 6eed0344734c98e4f07b42a491fda18bf403e18d
                                                Found module /usr/lib/x86_64-linux-gnu/libbrotlidec.so.1.0.9 with build-id: 9a0d0fa9c04097bef6c47d4cfd28781af365e387
                                                Found module /usr/lib/x86_64-linux-gnu/libbz2.so.1.0.8 with build-id: 2d7a13911b641918007f169632c07625b3079b69
                                                Found module /usr/lib/x86_64-linux-gnu/libgraphite2.so.3.2.1 with build-id: 37c7789f0441c6257ce781a9a5c5b85cabb75036
                                                Found module /usr/lib/x86_64-linux-gnu/libdatrie.so.1.4.0 with build-id: 8d8af19e535a4c2ab38b522997004fed3ddcbe87
                                                Found module /usr/lib/x86_64-linux-gnu/libgcrypt.so.20.2.10 with build-id: e0e59aebffb1577f553c1c942719cfb19ab505c8
                                                Found module /usr/lib/x86_64-linux-gnu/libcap.so.2.59 with build-id: a37e8eb37a7a87c8b610dbcd40e70320311a07c7
                                                Found module /usr/lib/x86_64-linux-gnu/liblz4.so.1.9.3 with build-id: 5782abf3e441c75c927fe36b5fd7c991d19434a3
                                                Found module /usr/lib/x86_64-linux-gnu/liblzma.so.5.2.5 with build-id: c67a0ef914a32749f75293e92f58739697278ffb
                                                Found module /usr/lib/x86_64-linux-gnu/libblkid.so.1.1.0 with build-id: d7ae760b43c4c8dece47ee88ca6edea64a9abed5
                                                Found module /usr/lib/x86_64-linux-gnu/libgmp.so.10.4.0 with build-id: 9da2b74e1d23a1cd358aaea3e8ec77a41b93034d
                                                Found module /usr/lib/x86_64-linux-gnu/libtasn1.so.6.6.2 with build-id: d885c98e00c53a8f3409c5aed9dcd407093e15ad
                                                Found module /usr/lib/x86_64-linux-gnu/libunistring.so.2.1.0 with build-id: 2bfaa7493a74c9d7c3868b328758ba8174948d4c
                                                Found module /usr/lib/x86_64-linux-gnu/libidn2.so.0.3.7 with build-id: d1c415a50f9a88dbf7dd0595571e3ef3abeab636
                                                Found module /usr/lib/x86_64-linux-gnu/libp11-kit.so.0.3.0 with build-id: 20c508416f2451b572a0cb85596ed6fa1f04eec4
                                                Found module /usr/lib/x86_64-linux-gnu/librt-2.33.so with build-id: 6b8b679b4a8af6f6b6b543205f65fb7f5be7393b
                                                Found module /usr/lib/x86_64-linux-gnu/libxcb-render.so.0.0.0 with build-id: cd99286d65beb0edba0fc9b965b88859962498e8
                                                Found module /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0 with build-id: c433630219a163d613d36297632847bcf012a56c
                                                Found module /usr/lib/x86_64-linux-gnu/libpng16.so.16.37.0 with build-id: fd5a56f12454013abf8ccf39d35b6bdd5afbc57b
                                                Found module /usr/lib/x86_64-linux-gnu/libfreetype.so.6.18.0 with build-id: a4f0772cc98e8f771cee05e5ddb7eef04712c021
                                                Found module /usr/lib/x86_64-linux-gnu/libfontconfig.so.1.12.0 with build-id: 0bd952349fa268687d073e9874f26e377607cf92
                                                Found module /usr/lib/x86_64-linux-gnu/libpixman-1.so.0.40.0 with build-id: 3e84cb961295d25c8862c6e1bc15decca91ad824
                                                Found module /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0.30000.0 with build-id: 46635986af5bea8b58a1fe13e5e9b0e228508696
                                                Found module /usr/lib/x86_64-linux-gnu/libthai.so.0.3.1 with build-id: f537d8b4cc2a809812ad375c9c80a371ac00e2f8
                                                Found module /usr/lib/x86_64-linux-gnu/libfribidi.so.0.4.0 with build-id: c7ea1c86203fda11e6720c0709aa895468b1bde2
                                                Found module /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.29 with build-id: a8e9ca6fd2dd26e1512bbe52d9a235d0676b4f7e
                                                Found module /usr/lib/x86_64-linux-gnu/libwayland-server.so.0.20.0 with build-id: fcd98f15ae7dab98deb8619141c516278de566d0
                                                Found module /usr/lib/x86_64-linux-gnu/libXrender.so.1.3.0 with build-id: f9fa50234e5f542c4e32fe3849078020f43f3261
                                                Found module /usr/lib/x86_64-linux-gnu/libXau.so.6.0.0 with build-id: efc46f16c741933e189bc363ee2acc635d90a9a5
                                                Found module /usr/lib/x86_64-linux-gnu/libsystemd.so.0.32.0 with build-id: dc96280ac576f313d0d96451fcbbb3ce76c07e27
                                                Found module /usr/lib/x86_64-linux-gnu/libresolv-2.33.so with build-id: b64eda64750f7d31d6fdf01641db827fb90bd0fa
                                                Found module /usr/lib/x86_64-linux-gnu/libmount.so.1.1.0 with build-id: a0fb05773e7f57659cbbc87651fe0b97a5bcb65d
                                                Found module /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0.6800.4 with build-id: b0cb2c219711592642dd3cec5be37f06083dfb77
                                                Found module /usr/lib/x86_64-linux-gnu/libz.so.1.2.11 with build-id: 487c492d01ffa699a2d81a2a66427f733c4c7777
                                                Found module /usr/lib/x86_64-linux-gnu/libgnutls.so.30.30.0 with build-id: 064b9bb6c2ef1461bcdf9f7164a5e98fe64e3dbc
                                                Found module /usr/lib/x86_64-linux-gnu/libplc4.so with build-id: 78f52c8786270b8a1f218c2a292fe042500e6e5c
                                                Found module /usr/lib/x86_64-linux-gnu/libplds4.so with build-id: 8a3cbab58732ff82a37ace1eb2e99c2e14ecb25c
                                                Found module /usr/lib/x86_64-linux-gnu/libffi.so.7.1.0 with build-id: cdff81e6273a31ae51cd175a164af0c783992fc5
                                                Found module /usr/lib/x86_64-linux-gnu/libpcre.so.1.2.13 with build-id: 1b1f91feb6837672c4292b2c3ff26bff0cc71d1a
                                                Found module /usr/lib/x86_64-linux-gnu/libfdk-aac.so.2.0.0 with build-id: d3b9cc416c3aae0286a17921330b16bca14e57a6
                                                Found module /usr/lib/x86_64-linux-gnu/libc-2.33.so with build-id: 76d1aab409296f4238aafd7e75342f8062578b8f
                                                Found module /usr/lib/x86_64-linux-gnu/libatspi.so.0.0.1 with build-id: 3ba1d909eeb6ee6dec719be2cfa9aea10b1a87fc
                                                Found module /usr/lib/x86_64-linux-gnu/libasound.so.2.0.0 with build-id: e1efa324d44b121a067532eae1c92e6ddd78f693
                                                Found module /usr/lib/x86_64-linux-gnu/libcairo.so.2.11600.0 with build-id: f6062387998b075aa74abbca5175d983c7bc5d6a
                                                Found module /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0.4800.11 with build-id: 5b39fca165463dfc08722b0cf532a07fb6acd9cb
                                                Found module /usr/lib/x86_64-linux-gnu/libwayland-client.so.0.20.0 with build-id: edfc146d7c582464db2197b9c51747b53f77b080
                                                Found module /usr/lib/x86_64-linux-gnu/GL/default/lib/libgbm.so.1.0.0 with build-id: b22bc3a20d8ea4a666e776eba94874e805c120ed
                                                Found module /usr/lib/x86_64-linux-gnu/libpipewire-0.3.so.0.335.0 with build-id: 64d2ca3f04bb7410ca260d07350835f94c16f52e
                                                Found module /usr/lib/x86_64-linux-gnu/libXrandr.so.2.2.0 with build-id: 7b02e00bcb811387f734c2444e812d85c03b452e
                                                Found module /usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0 with build-id: adcda11d008e5e0dbf5c5d8af031bae14f813a6c
                                                Found module /usr/lib/x86_64-linux-gnu/libXext.so.6.4.0 with build-id: 9f6b23c7454b760b57c94f5e8f4d0dbe76d909b1
                                                Found module /usr/lib/x86_64-linux-gnu/libXdamage.so.1.1.0 with build-id: ccaf00ddfd2af2b5ddfca52df5ad908b339648f2
                                                Found module /usr/lib/x86_64-linux-gnu/libXcomposite.so.1.0.0 with build-id: 0adf2de4b14f1732b599fcbae7fa24ab1f3a5a55
                                                Found module /usr/lib/x86_64-linux-gnu/libX11.so.6.4.0 with build-id: 51f60f9eeac221fc3445ce570e6fb6baa83e604a
                                                Found module /usr/lib/x86_64-linux-gnu/libm-2.33.so with build-id: 582330fbfef5642cd490fd6a1fb780396be9a758
                                                Found module /usr/lib/x86_64-linux-gnu/libxkbcommon.so.0.0.0 with build-id: fc58e7701adc7efa1b378075dad594e6f47fc9f6
                                                Found module /usr/lib/x86_64-linux-gnu/libxcb.so.1.1.0 with build-id: 7d67eabc92da2fbd05cb69b339e41cea79f9fa85
                                                Found module /usr/lib/x86_64-linux-gnu/libexpat.so.1.8.6 with build-id: e60fdb93352dc5c5e41aa4161b8c906a4eac85b7
                                                Found module /usr/lib/x86_64-linux-gnu/libdbus-1.so.3.19.13 with build-id: 0b00b3cd752bc63d40f300a691e2840f8911f3f2
                                                Found module /usr/lib/x86_64-linux-gnu/libdrm.so.2.4.0 with build-id: 02a0c796e3aec09632fd8e742941f97723e089e3
                                                Found module /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0.6800.4 with build-id: e37f2ea1c8fd9e1caba0908592a77a10e327efc0
                                                Found module /usr/lib/x86_64-linux-gnu/libcups.so.2 with build-id: 532f7c3d706769ad755efba89daba1e6e9e970c3
                                                Found module /usr/lib/x86_64-linux-gnu/libatk-bridge-2.0.so.0.0.0 with build-id: 62428b9d3948076922c2d899cd327047facd0c51
                                                Found module /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0.23609.1 with build-id: d81312a37ac277e52927c1d5a4051d54e253d43f
                                                Found module /usr/lib/x86_64-linux-gnu/libnspr4.so with build-id: 5d058bfb3f386e4a12e18fc01e1fd804e1e42857
                                                Found module /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.6800.4 with build-id: da7b797433b63ad34fcf809ec582fc02510babb4
                                                Found module /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.6800.4 with build-id: 691d41b4a41490d4fd314b4383949d04372f53b9
                                                Found module /usr/lib/x86_64-linux-gnu/libpthread-2.33.so with build-id: a25ab06ac2795c4cb998ae8560caa564bf67d371
                                                Found module /usr/lib/x86_64-linux-gnu/libdl-2.33.so with build-id: d9535f8920ca1604fe08fa23cbf35691f74dc4d4
                                                Found module /app/chromium/nonfree-codecs/lib/libffmpeg.so with build-id: b385036988e23f12da9f3eb212da777af70c4766
                                                Found module /usr/lib/x86_64-linux-gnu/ld-2.33.so with build-id: f9952eb3aa68da1da20a6a5c1a8e608cc56b9e82
                                                Found module linux-vdso.so.1 with build-id: 3eaf1487ed59834c6f348a7df2e791f82dbedfc2
                                                Stack trace of thread 3042:
                                                #0  0x000055988c52224e n/a (/app/chromium/chrome + 0x9d3d24e)
                                                #1  0x000055988c522003 n/a (/app/chromium/chrome + 0x9d3d003)
                                                #2  0x000055988c772338 n/a (/app/chromium/chrome + 0x9f8d338)
                                                #3  0x000055988c6c3c59 n/a (/app/chromium/chrome + 0x9edec59)
                                                #4  0x00005598892707f4 n/a (/app/chromium/chrome + 0x6a8b7f4)
                                                #5  0x0000559889270bac n/a (/app/chromium/chrome + 0x6a8bbac)
                                                #6  0x0000559884936b00 n/a (/app/chromium/chrome + 0x2151b00)
                                                #7  0x0000559884935a9a n/a (/app/chromium/chrome + 0x2150a9a)
                                                #8  0x0000559884cb4d99 n/a (/app/chromium/chrome + 0x24cfd99)
                                                #9  0x000055988625e405 n/a (/app/chromium/chrome + 0x3a79405)
                                                #10 0x000055988625da3a n/a (/app/chromium/chrome + 0x3a78a3a)
                                                #11 0x000055988625d878 n/a (/app/chromium/chrome + 0x3a78878)
                                                #12 0x0000559886107f90 n/a (/app/chromium/chrome + 0x3922f90)
                                                #13 0x0000559885f90903 n/a (/app/chromium/chrome + 0x37ab903)
                                                #14 0x00007fad0c1eb3ba n/a (/usr/lib/x86_64-linux-gnu/libpthread-2.33.so + 0x83ba)
                                                #15 0x00007fad0af8eb03 n/a (/usr/lib/x86_64-linux-gnu/libc-2.33.so + 0x101b03)
Mar 04 11:35:53 fedora systemd[1]: systemd-coredump@2-92650-0.service: Deactivated successfully.
Mar 04 11:35:53 fedora audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=systemd-coredump@2-92650-0 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
Mar 04 11:35:53 fedora systemd[1]: systemd-coredump@2-92650-0.service: Consumed 3.321s CPU time.
Mar 04 11:35:53 fedora audit: BPF prog-id=0 op=UNLOAD
Mar 04 11:35:53 fedora audit: BPF prog-id=0 op=UNLOAD
Mar 04 11:35:53 fedora audit: BPF prog-id=0 op=UNLOAD
Mar 04 11:35:54 fedora abrtd[1286]: Size of '/var/spool/abrt' >= 5000 MB (MaxCrashReportsSize), deleting old directory 'ccpp-2022-03-03-15:08:23.33248-10116'
Mar 04 11:35:54 fedora abrt-server[92667]: Unsupported container technology
Mar 04 11:35:54 fedora abrt-server[92667]: Lock file '.lock' was locked by process 92670, but it crashed?
Mar 04 11:35:54 fedora abrt-notification[92731]: Process 52513 (chrome) crashed in ??()
Mar 04 11:36:25 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9019 error:1 (Permission denied)
Mar 04 11:36:26 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9023 error:1 (Permission denied)
Mar 04 11:36:30 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9052 error:1 (Permission denied)
Mar 04 11:36:40 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9153 error:1 (Permission denied)
Mar 04 11:36:41 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9157 error:1 (Permission denied)
Mar 04 11:36:51 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9238 error:1 (Permission denied)
Mar 04 11:37:03 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9344 error:1 (Permission denied)
Mar 04 11:37:18 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9476 error:1 (Permission denied)
Mar 04 11:37:31 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9585 error:1 (Permission denied)
Mar 04 11:37:46 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9717 error:1 (Permission denied)
Mar 04 11:38:15 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:9959 error:1 (Permission denied)
Mar 04 11:38:28 fedora pipewire-pulse[2597]: mod.protocol-pulse: client 0x5612c8e937f0 [Chromium input]: ERROR command:38 (SET_SOURCE_VOLUME) tag:10065 error:1 (Permission denied)
```
</details>

System info:
* chromium version: 99.0.4844.51
* OS: Fedora Linux 35
* Kernel: 5.16.11-200.fc35.x86_64

**Comment by refi64** [comment]
Chromium 99 tarballs disappeared and I wasn't sure if it was a glitch or a revert...well, I guess it was a revert...

**Comment by refi64** [comment]
Should be fixed in #206.

**Comment by christianfosli** [comment]
Thanks :smile:
Closing.
